### DISCUSSION TITLE
MiniRocket .fit() OOM issue with large datasets

### DISCUSSION THREAD

**Discussion by Arderiu** [timeseriesAI/tsai#743] [Q&A] [answered]
I have one question regarding MiniRocket: I have a training dataset with size (13604, 1, 3000), and when I call the `` mrf.fit(X_train)`` I get a OOM error. A solution I found is to just fit with a subset of the training set, e.g. `` mrf.fit(X_train13000:,:,])``. I assume, this will negatively impact the performance of the feature computation... Is there any way to fit the model "batch by batch" to avoid OOM issues? Thanks!

**Comment by oguiza** [comment, marked as answer]
No, there isn't @Arderiu.
Actually, the authors of the paper recommend using MiniRocketClassifier or MiniRocketRegressor when the dataset is < 10k samples.
However, there's a Minirocket version written in Pytorch that can be used with any dataset size. 
You can find more information in this [tutorial](https://github.com/timeseriesAI/tsai/blob/main/tutorial_nbs/10_Time_Series_Classification_and_Regression_with_MiniRocket.ipynb).

**Reply by RoyiAvital** [reply]
Is there a way to wrap a PyTorch data loader?
In all examples it uses `X, y` which are NumPy.

**Reply by oguiza** [reply]
I don't understand what you mean by "wrap a Pytorch data loader".

**Reply by RoyiAvital** [reply]
Use a `PyTorch` data loader instead of a NumPy array.

**Reply by oguiza** [reply]
Yes, you can use MiniRocket or MiniRocketPlus (https://timeseriesai.github.io/tsai/models.minirocketplus_pytorch.html) with a regular Pytorch dataloader like any other model in tsai.
